### DISCUSSION TITLE
Can I simply create new components?

### DISCUSSION THREAD

**Discussion by Suolucidir** [quasarframework/quasar#12600] [Quasar App Extensions]
Hi All,

I used a previous version of quasar where I could simply add a new component in the components folder (src\components), give it a name in the exports default, import it in (another) form component, define it as one of the imported components and use it. That form component was imported in the src/router/index.js file where I defined it as the component for the root path(/).

It seemed straight-forward way to add my own version of a component to my solution.

I am totally confused by what I have to do now. 
Do I have to use app-extensions now? (I don't want to publish anything to the NPM registry). 
Or am I looking at the wrong path altogether?

Thanks!

**Comment by smolinari** [comment]
> Can I simply create new components?

Yes. What problem are you running in to?

Scott

**Comment by Suolucidir** [comment]
Ok then - how? What is the simplest fastest way to do it please? Can I do it simply like it's explained in vuejs documentation or do I need an App Extension? Thanks.

**Reply by smolinari** [reply]
You don't necessarily need an app extension. You simply create your component and call it into another component and use it, just like any other Vue or Quasar component. 

Scott

**Comment by yusufkandemir** [comment]
When you create a Quasar CLI project, it comes with an example layout component(_`layouts/MainLayout.vue`_), a page component(_`pages/Index.vue`_), and some partial components(_e.g. `components/EssentialLink.vue`_), and an example parent-child route configuration in `router/index.js`. Creating components, configuring routes, etc. is explained in Quasar, Vue, Vue Router, etc. docs. So, I couldn't understand what is confusing, to be honest.
